src/test/kotlin/com/example/passmanager/repositories/PassRepositoryImplTest.kt

```diff
@@ -11,207 +10,253 @@ import org.assertj.core.api.Assertions.assertThat
 import org.bson.types.ObjectId
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.boot.test.context.SpringBootTest
-import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate
 import org.springframework.data.mongodb.core.exists
+import org.springframework.data.mongodb.core.findById
 import org.springframework.data.mongodb.core.query.Criteria.where
 import org.springframework.data.mongodb.core.query.Query.query
 import org.springframework.data.mongodb.core.query.isEqualTo
+import reactor.kotlin.test.test
 import java.math.BigDecimal
 import java.time.LocalDate
 import java.time.ZoneId
 import kotlin.test.Test
-import kotlin.test.assertEquals
 import kotlin.test.assertFalse
-import kotlin.test.assertTrue
+import kotlin.test.assertNotNull
 
 @SpringBootTest
 @WithMongoTestContainer
 class PassRepositoryImplTest {
 
     @Autowired
-    private lateinit var mongoTemplate: MongoTemplate
+    private lateinit var mongoTemplate: ReactiveMongoTemplate
 
     @Autowired
     private lateinit var passRepository: PassRepository
 
     @Test
     fun `finding pass by existing id should return pass by id`() {
         // GIVEN
-        val inserted = mongoTemplate.insert(passToCreate)
+        val insertedPass = mongoTemplate.insert(passToCreate)
 
         // WHEN
-        val passById = passRepository.findById(inserted.id.toString())
+        val passById = insertedPass
+            .mapNotNull { it.id.toString() }
+            .flatMap { insertedPassId -> passRepository.findById(insertedPassId) }
 
         // THEN
-        assertThat(passById?.id).isNotNull().isEqualTo(inserted.id)
+        passById.test()
+            .assertNext { assertNotNull(it) }
```

Comment:
Why assertNotNull?
We want to test that returned pass is precisely our inserted pass